pythonosc/osc_tcp_server.py

```diff
@@ -0,0 +1,392 @@
+            for r in resp:
+                if r is not None:
+                    if not isinstance(r, list):
+                        r = [r]
+                    msg = osc_message_builder.build_msg(r[0], r[1:])
+                    b = struct.pack('!I', len(msg.dgram))
+                    self.request.sendall(b + msg.dgram)
+
+    def recvall(self, count):
+        buf = b''
+        while count > 0:
+            newbuf = self.request.recv(count)
+            if not newbuf:
+                return None
+            buf += newbuf
+            count -= len(newbuf)
+        return buf
+
+
+class _TCPHandler1_1(socketserver.BaseRequestHandler):
+    """Handles correct OSC1.1 messages.
+
+    Whether this will be run on its own thread, the server's or a whole new
+    process depends on the server you instantiated, look at their documentation.
+
+    This method is called after a basic sanity check was done on the datagram,
+    basically whether this datagram looks like an osc message or bundle,
+    if not the server won't even bother to call it and so no new
+    threads/processes will be spawned.
+    """
+    def handle(self):
+        LOG.debug("handle OSC 1.1 protocol")
+        while True:
+            packets = self.recvall()
+            if packets is None:
+                break
+
+            for p in packets:
+                resp = _call_handlers_for_packet(p, self.server.dispatcher)
+                for r in resp:
+                    if not isinstance(r, list):
+                        r = [r]
+                    msg = osc_message_builder.build_msg(r[0], r[1:])
+                    self.request.sendall(slip.encode(msg.dgram))
+
+    def recvall(self):
+        buf = self.request.recv(4096)
+        if not buf:
+            return None
+        # If the last byte is not an END marker there could be more data coming
+        while buf[-1] != 192:
+            newbuf = self.request.recv(4096)
+            if not newbuf:
+                # Maybe should raise an exception here?
+                break
+            buf += newbuf
+
+        packets = [slip.decode(p) for p in buf.split(slip.END_END)]
+        return packets
+
+
+class OSCTCPServer(socketserver.TCPServer):
+    """Superclass for different flavors of OSCTCPServer"""
+
+    def __init__(self, server_address, dispatcher, mode: str = MODE_1_1):
+        self.request_queue_size = 300
+        self.mode = mode
+        if mode not in [MODE_1_0, MODE_1_1]:
+            raise ValueError("OSC Mode must be '1.0' or '1.1'")
+        if self.mode == MODE_1_0:
+            super().__init__(server_address, _TCPHandler1_0)
+        else:
+            super().__init__(server_address, _TCPHandler1_1)
+        self._dispatcher = dispatcher
+
+    def verify_request(self, request, client_address):
+        """Returns true if the data looks like a valid OSC TCP datagram."""
+        # d = request.recv(9999).decode("utf-8")
```

Comment:
remove commented out lines/fix method description or remove method altogether if it's unused?

Reply:
Removed unused method